packages/utils/tokens/src/text-tokens.ts

```diff
@@ -1,15 +1,34 @@
-import { TextStyle } from 'react-native';
-import { IOperationSet } from '@uifabricshared/foundation-tokens';
-import { ITheme } from '@uifabricshared/theming-ramp';
-
-export interface ITextTokens {
-  fontFamily?: TextStyle['fontFamily'] | string;
-  fontSize?: TextStyle['fontSize'] | string;
-  fontWeight?: TextStyle['fontWeight'] | string;
+import { TextStyle, TextProps } from 'react-native';
+import { ITheme, ITypography } from '@uifabricshared/theming-ramp';
+import { styleFunction } from '@uifabricshared/foundation-tokens';
+
+export interface ITextVariantTokens {
+  fontVariant?: keyof ITypography['variants'] | string;
+}
+
+export interface ITextStyleTokens {
+  fontFamily?: keyof ITypography['families'] | TextStyle['fontFamily'] | string;
+  fontSize?: keyof ITypography['sizes'] | TextStyle['fontSize'] | string;
+  fontWeight?: keyof ITypography['weights'] | TextStyle['fontWeight'] | string;
+}

```

Comment:
I'm not sure we want | string on all of these types?  Unless we want to do something similar to the colors object where we also make it a dictionary so that variants could be extended for a subtheme

Reply:
which might actually be a good idea

Reply:
So, should these be kept as is? Or is there additional work needed to make it behave like the colors object?

Reply:
Actually isn't the string part redundant?  I would expect that it picks up string from TextStyle['fontFamily'] and string | number from TextStyle['fontSize'].  Not blocking but good to test/update at some point.